src/kern/npf_inet.c

```diff
@@ -623,6 +623,11 @@ npf_cache_all(npf_cache_t *npc)
 		    sizeof(struct icmp6_hdr));
 		l4flags = NPC_LAYER4 | NPC_ICMP;
 		break;
+	case IPPROTO_GRE:
+		/* Cache: layer 4 - GRE. */
+		npc->npc_l4.hdr = nbuf_advance(nbuf, hlen, npf_pptp_gre_hdr_len());
```

Comment:
So, this checks for the `pptp_gre_hdr_t` if the protocol is GRE. Questions:
- As I understand, PPTP GRE and standard GRE headers are different. Moreover, they are incompatible, right?
- How can we distinguish between the GRE and PPTP GRE?
- Can this check and the PPTP ALG code can somehow interfere with the standard GRE packets?

Reply:
You are right. They are different. PPTP protocol uses Enhanced GRE while other application could use a standard GRE header.
There is a field that indicated GRE version.
This check could try to nbuf_ensure_contig() enhanced GRE header for a packet with
a standard GRE header which could be smaller. So, I think I could add an additional check to deal only with enhanced GRE headers.